libs/notification/src/services/notification.service.ts

```diff
@@ -0,0 +1,51 @@
+import {Inject, Injectable} from '@angular/core';
+import {fromEvent, Observable, throwError} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
+
+import {NOTIFICATION_SUPPORT} from '../tokens/support';
+
+@Injectable({
+    providedIn: 'root',
+})
+export class NotificationService {
+    private readonly notSupportedError$ = throwError(
+        () => new Error('Notification API is not supported in your browser'),
+    );
+
+    constructor(@Inject(NOTIFICATION_SUPPORT) private readonly support: boolean) {}
+
+    requestPermission(): Observable<NotificationPermission> {
+        if (!this.support) {
+            return this.notSupportedError$;
+        }
+
+        /**
+         * TODO: replace deprecated callback with promise after Safari 15+ support
```

Comment:
Safari 16 supports Permissions API so I think we can just keep it here until 16 and drop this method completely after that.